src/components/SingleArticle.jsx

```diff
@@ -11,19 +11,39 @@ function SingleArticle() {
   const [article, setArticle] = useState({});
   const [isLoading, setIsLoading] = useState(true);
   const [isError, setIsError] = useState(false);
+  const [votesCount, setVotesCount] = useState(0);
 
   useEffect(() => {
     setIsLoading(true);
     fetchArticle(articleId)
       .then((fetchedArticle) => {
         setArticle(fetchedArticle);
+        setVotesCount(fetchedArticle.votes);
         setIsLoading(false);
       })
       .catch((error) => {
         setIsLoading(false);
         setIsError(true);
       });
-  }, [])
+  }, []);
+
+  function handleClickIncrement() {
+    patchArticleVotes(articleId, 1).catch(() => {
+      setVotesCount((currentVotesCount) => {
+        return currentVotesCount - 1;
+      });
+    });
+    setVotesCount((currentVotesCount) => currentVotesCount + 1);
+  }
+
```

Comment:
Both of these functions essentially do the same thing...

They both call the patchArticleVotes function with an id and a number.

Can you refactor it to be dry and reusable?